### PR TITLE
BREAKING: Require snaps to request at least one handler permission

### LINKED ISSUE #899: Snaps with 0 permissions shouldn't crash the extension
Snaps without any permissions don't make a lot of sense once we do #855.

An edge-case for snaps with 0 permissions should be handled. How it's handled is a matter of debate. For now let's go with allowing 0-permission snaps due to less work needed and more aligned with the permissionless ethos.

Solution: Disallow snaps with 0 permissions. Snaps without permissions cannot do anything.

### Outcome
- Fix for 0-permission snaps crashing the extension.
- E2E test for installing such a snap.

### PR DESCRIPTION
**Pull request by FrederikBolding**
Require snaps to request at least one handler permission. 

Fixes #899

### CODE REVIEW

File: packages/snaps-controllers/src/snaps/SnapController.ts [hunk 1]
```diff
@@ -2292,15 +2292,27 @@ export class SnapController extends BaseController<
...
-      }
+    const permissionKeys = Object.keys(processedPermissions);
+
+    const handlerPermissions = Object.values(handlerEndowments);
+
+    assert(
+      permissionKeys.some((key) => handlerPermissions.includes(key)),
+      `A snap must request at least one of the following permissions: ${handlerPermissions.join(
+        ', ',
+      )} `,
```
Comment (Mrtenz) [inline packages/snaps-controllers/src/snaps/SnapController.ts:2303]:
```suggestion
      )}.`,
```

File: packages/snaps-utils/src/manifest/validation.ts [hunk 2]
```diff
@@ -105,6 +105,8 @@ export const PermissionsStruct = type({
     object({ jobs: CronjobSpecificationArrayStruct }),
   ),
   'endowment:rpc': optional(RpcOriginsStruct),
+  snap_dialog: optional(object({})),
+  // TODO: Remove
```
Comment (Mrtenz) [inline packages/snaps-utils/src/manifest/validation.ts:109]: ?

**Review by Mrtenz — CHANGES_REQUESTED** [review]

File: packages/snaps-controllers/src/snaps/SnapController.ts (hunk 1 above)
Comment (Mrtenz) [reply packages/snaps-controllers/src/snaps/SnapController.ts:2303]: Update the test too.

Comment (FrederikBolding) [reply packages/snaps-controllers/src/snaps/SnapController.ts]: done

**Review by Mrtenz — DISMISSED** [review]

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**